tests/src/test_cases/test_anim_timeline.c

```diff
@@ -322,4 +327,99 @@ void test_anim_timeline_reverse(void)
     TEST_ASSERT_EQUAL(20, lv_obj_get_x(obj));
 }
 
+
+void anim1_exec_cb(void * var, int32_t v)
+{
+    LV_UNUSED(var);
+    printf("[anim1] %d\n", v);
+}
+void anim1_start(lv_anim_t * a)
+{
+    LV_UNUSED(a);
+    printf("[anim1] start\n");
+    anim1_start_called++;
+}
+void anim1_completed(lv_anim_t * a)
+{
+    LV_UNUSED(a);
+    printf("[anim1] completed\n");
+    anim1_completed_called++;
+}
+
+void anim2_exec_cb(void * var, int32_t v)
+{
+    LV_UNUSED(var);
+    printf("                    [anim2] %d\n", v);
+}
+void anim2_start(lv_anim_t * a)
+{
+    LV_UNUSED(a);
+    printf("                    [anim2] start\n");
+    anim2_start_called++;
+}
+void anim2_completed(lv_anim_t * a)
+{
+    LV_UNUSED(a);
+    printf("                    [anim2] completed\n");
+    anim2_completed_called++;
+}
+
+
+void test_anim_timeline_with_anim_start_cb_and_completed_cb(void)
+{
+    lv_anim_t anim1;
+    lv_anim_t anim2;
+
+    lv_anim_init(&anim1);
+    lv_anim_set_values(&anim1, 0, 1024);
+    lv_anim_set_duration(&anim1, 300);
+    lv_anim_set_exec_cb(&anim1, anim1_exec_cb);
+    lv_anim_set_start_cb(&anim1, anim1_start);
+    lv_anim_set_completed_cb(&anim1, anim1_completed);
+    anim1_start_called = 0;
+    anim1_completed_called = 0;
+
+    lv_memcpy(&anim2, &anim1, sizeof(anim1));
+    lv_anim_set_duration(&anim2, 300);
+    lv_anim_set_path_cb(&anim2, lv_anim_path_ease_in);
+    lv_anim_set_exec_cb(&anim2, anim2_exec_cb);
+    lv_anim_set_start_cb(&anim2, anim2_start);
+    lv_anim_set_completed_cb(&anim2, anim2_completed);
+    anim2_start_called = 0;
+    anim2_completed_called = 0;
+
+    lv_anim_timeline_t * timeline = lv_anim_timeline_create();
+    lv_anim_timeline_add(timeline,   0, &anim1);
+    lv_anim_timeline_add(timeline, 200, &anim2);
+    lv_anim_timeline_start(timeline);
+
+
+    /*
+     *   |-----anim1-----|
+     *           |----anim2----|
+     *
+     *   0       200   300    500
+     */
+
+    lv_test_wait(20); /*Wait 20 ms*/
+    TEST_ASSERT_EQUAL(1, anim1_start_called);
+    TEST_ASSERT_EQUAL(0, anim1_completed_called);
+    TEST_ASSERT_EQUAL(0, anim2_start_called);
+    TEST_ASSERT_EQUAL(0, anim2_completed_called);
+
+    lv_test_wait(200); /*Now we are at 220ms */
+    TEST_ASSERT_EQUAL(1, anim1_start_called);
+    TEST_ASSERT_EQUAL(0, anim1_completed_called);
+    TEST_ASSERT_EQUAL(1, anim2_start_called);
+    TEST_ASSERT_EQUAL(0, anim2_completed_called);
+
+    lv_test_wait(100); /*Now we are at 320ms */
+    TEST_ASSERT_EQUAL(1, anim1_start_called);
+    TEST_ASSERT_EQUAL(1, anim1_completed_called);
+    TEST_ASSERT_EQUAL(1, anim2_start_called);
+    TEST_ASSERT_EQUAL(0, anim2_completed_called);
```

Comment:
Please test at >500ms too, when all animations should be finished.

Reply:
When I test with the Serial monitor 
or when using it in various projects, 
the  `start_cb` and `completed_cb` function appears to work.

And when I tested it with `test_anim_timeline.c` with the last update, 
it passed the test.

![image](https://github.com/lvgl/lvgl/assets/15370479/fab6c89a-9b79-41ed-98dd-8b942b2db3ff)

However, when `test_anim_timeline.c` had the addition:

```
lv_test_wait(200); /*Now we are at 520ms */
```

added and then tested, 
an error occurred during testing. 

![image](https://github.com/lvgl/lvgl/assets/15370479/554b7ab9-8c17-4fcc-b4b0-319d841d400c)

I don't understand why an error occurs 
during testing when an additional 200ms wait 
by `lv_test_wait(200);` is added at the end.

Reply:
If you scroll up in the log you will see exactly where it fails. You can also push the latest test code so that I can also try it out.

Reply:
> If you scroll up in the log you will see exactly where it fails. 
> You can also push the latest test code so that I can also try it out.

It seems that testing `test_anim_timeline.c` on Windows 
may result in memory leaks.

![image](https://github.com/lvgl/lvgl/assets/15370479/b674cb0f-d5e3-497a-9a5a-d5eca70d5ef7)